### PR TITLE
Keep solr fields encapsulated within the SolrDocument

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: app/models/concerns/curation_concerns/solr_document_behavior.rb [hunk 1]
```diff
@@ -29,12 +29,12 @@ def to_model
...
 
     def collection?
-      hydra_model == 'Collection'
+      hydra_model == Collection
     end
 
     # Method to return the ActiveFedora model
     def hydra_model
-      self[Solrizer.solr_name('active_fedora_model', Solrizer::Descriptor.new(:string, :stored, :indexed))]
+      fetch(Solrizer.solr_name('has_model', :symbol)).first.constantize
```
Comment (mjgiarlo) [inline app/models/concerns/curation_concerns/solr_document_behavior.rb]: Why the change from `active_fedora_model` to `has_model`?

Comment (jcoyne) [inline app/models/concerns/curation_concerns/solr_document_behavior.rb]: Because `has_model_ssim` is used throughout the stack. https://github.com/projecthydra/active_fedora/pull/1093

Comment (mjgiarlo) [inline app/models/concerns/curation_concerns/solr_document_behavior.rb]: OK
